src/Microsoft.AspNet.Security.OAuth/OAuthBearerAuthenticationMiddleware.cs

```diff
@@ -2,21 +2,25 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using Microsoft.AspNet.Builder;
-using Microsoft.AspNet.Security.DataHandler;
 using Microsoft.AspNet.Security.DataProtection;
 using Microsoft.AspNet.Security.Infrastructure;
 using Microsoft.Framework.Logging;
 using Microsoft.Framework.OptionsModel;
+using Microsoft.IdentityModel.Protocols;
 using System;
+using System.Collections.ObjectModel;
+using System.Diagnostics.CodeAnalysis;
+using System.IdentityModel.Tokens;
+using System.Net.Http;
 
 namespace Microsoft.AspNet.Security.OAuth
 {
-    /// <summary>
-    /// Bearer authentication middleware component which is added to an HTTP pipeline. This class is not
-    /// created by application code directly, instead it is added by calling the the IAppBuilder UseOAuthBearerAuthentication
-    /// extension method.
-    /// </summary>
-    public class OAuthBearerAuthenticationMiddleware : AuthenticationMiddleware<OAuthBearerAuthenticationOptions>
+	/// <summary>
+	/// Bearer authentication middleware component which is added to an HTTP pipeline. This class is not
+	/// created by application code directly, instead it is added by calling the the IAppBuilder UseOAuthBearerAuthentication
+	/// extension method.
+	/// </summary>
+	public class OAuthBearerAuthenticationMiddleware : AuthenticationMiddleware<OAuthBearerAuthenticationOptions>
```

Comment:
It looks like you've merged a lot of the old AAD and ADFS scenarios down into this component. Can is still be used as a simple validator without any of the metdata stuff? E.g. can I still just provide my own token validator without setting any of the metadata, configuration, or backchannel properties? It looks like it will always try to set up the configuration manager.